### ISSUE TITLE
ChaCha20 + AES-KDF won't open in Tusk

### ISSUE THREAD

**Issue by Gitii** [subdavis/Tusk#101]
## This issue is a 

- [x] bug
- [ ] feature
- [ ] question

### Please describe the current behavior, and explain why it's bad.

If the masterpassword contains an '@'-character the unlock fails.
To double check, I typed the password in notepad, copy and pasted it in both
- keepass -> worked
- tusk -> failed

Creating an new safe with the same password except an '@' -> works in tusk, too.

### Please describe how you think it should change.

Please, check if '@' is somehow handled specially by tusk.

### Anything else?

This issue seems weird, but nevertheless it does not work for me.

Greatings,

Gitii

**Comment by subdavis** [comment]
Hey there, @Gitii 

This issue has been pestering me for a while because I've seen several reports but cannot reproduce.  

Could you provide a few more details:

* what OS?
* What browser?
* Can you create a new safe with the same password including the '@' and test it?  I'm worried this is a subtle crypto bug.
* Can you test the same things on KeeWeb.info ?  We use the same crypto engine, and I fear it may be a bug there.

Thank you!!

**Comment by Gitii** [comment]
You are right. It could be in combination with certain password safes.
I created a new safe (default creation settings) with '@' and it works!
But if I create another one and choose **ChaCha20**, it fails.

My tests are not bullet proof, I did not test all combinations.
I uploaded two test databases (password is 'a@b'):
* test_at -> default settings -> works
* test_at2 -> ChaCha2 -> fails

Both can be openend by keepass2
But not with tusk and keepass.info.

but:
* test_at3 -> ChaCha2  and password 'ab' -> **fails**, too

https://ufile.io/1ilkb (valid for 7 days)



Tomorrow, I will try and verify it at work with my password safe.
In addition I will check the database settings.

My specs are:
Windows 10 1709,
Vivaldi 1.14.1077.45 (based on Chrome 64.0.3282.143)
